app/models/subnet.rb

```diff
@@ -152,12 +152,16 @@ def cidr=(cidr)
     nil
   end
 
-  def supports_ipam_mode?(mode)
-    self.class.supported_ipam_modes.include?(mode)
+  def self.supports_ipam_mode?(mode)
+    supported_ipam_modes.include?(mode)
+  end
+
+  def self.supports_dhcp?
+    supports_ipam_mode?(:dhcp)
   end
 
   def dhcp?
-    !!(dhcp and dhcp.url and !dhcp.url.blank?)
+    !!(dhcp && dhcp.url && !dhcp.url.blank? && self.class.supports_dhcp?)
```

Comment:
I think this expression should check for `supports_dhcp?` first, the rest doesn't make sense if that one is false. Also, checking for `.url` and `!url.blank?` is equivalent to `url.present?` (I know it's not your fault here but hey :smile: )

```ruby
self.class.supports_dhcp? && dhcp && dhcp.url.present?
```
I don't think the `!!` is needed here, it'll always be a boolean expression